code/procs/jobprocs.dm

```diff
@@ -704,6 +704,16 @@ else if (istype(JOB, /datum/job/security/security_officer))\
 			trinket = new/obj/item/reagent_containers/food/snacks/ingredient/egg/bee/buddy(src)
 		else
 			trinket = new/obj/item/reagent_containers/food/snacks/ingredient/egg/bee(src)
+	else if (src.traitHolder && src.traitHolder.hasTrait("petperson"))
+		var/obj/item/pet_carrier/carrier = new/obj/item/pet_carrier(src)
+		var/picked = pick(filtered_concrete_typesof(/mob/living/critter/small_animal/, GLOBAL_PROC_REF(filter_carrier_pets)))
+		var/mob/living/critter/small_animal/pet = new picked(src)
+		pet.ai_type = /datum/aiHolder/wanderer
+		pet.aggressive = FALSE
+		pet.randomize_name()
+		pet.ai_retaliate_persistence = RETALIATE_ONCE
+		carrier.trap_mob(pet, src)
+		src.force_equip(carrier, SLOT_IN_BACKPACK, TRUE)
```

Comment:
Pet carriers don't actually fit in backpacks so it's a little strange that it starts there, also some jobs (clown) don't start with a backpack. Probably better to just go `l_hand` -> `r_hand` -> floor, I don't think there are any jobs that spawn with two items in hand, but there might be other trait combinations.

Reply:
Oh that's a goof, I was so sure they fit in backpacks